h/storage.py

```diff
@@ -187,9 +184,16 @@ def update_annotation(session, id_, data):
     # annotation object.
     document = data.pop('document', None)
 
-    annotation = session.query(models.Annotation).get(id_)
+    annotation = request.db.query(models.Annotation).get(id_)
     annotation.updated = updated
 
+    group = group_service.find(annotation.groupid)
+    if group is None:
+        raise schemas.ValidationError('group: ' +
```

Comment:
Since the annotation has already been saved in the DB and we checked the groupid on creation, presumably this should never happen. However, since the constraint is not enforced at the database level in the current schema it doesn't hurt to check.